pkg/vm/builtins/keccak.go

```diff
@@ -0,0 +1,257 @@
+package builtins
+
+import (
+	"encoding/binary"
+	"errors"
+	"fmt"
+
+	"github.com/NethermindEth/cairo-vm-go/pkg/utils"
+	"github.com/NethermindEth/cairo-vm-go/pkg/vm/memory"
+	"github.com/consensys/gnark-crypto/ecc/stark-curve/fp"
+
+	"github.com/holiman/uint256"
+	"golang.org/x/crypto/sha3"
+)
+
+const KeccakName = "keccak"
+const cellsPerKeccak = 16
+const inputCellsPerKeccak = 8
+
+type Keccak struct {
+}
+
+func (k *Keccak) CheckWrite(segment *memory.Segment, offset uint64, value *memory.MemoryValue) error {
+	return nil
+}
+
+func (k *Keccak) InferValue(segment *memory.Segment, offset uint64) error {
+	hashIndex := offset % cellsPerKeccak
+	if hashIndex < inputCellsPerKeccak {
+		return errors.New("cannot infer value")
+	}
+
+	startOffset := offset - hashIndex
+	var data [200]byte
+	for i := uint64(0); i < inputCellsPerKeccak; i++ {
+		value := segment.Peek(startOffset + i)
+		if !value.Known() {
+			return fmt.Errorf("cannot infer value: input value at offset %d is unknown", startOffset+i)
+		}
+		v, err := value.FieldElement()
+		if err != nil {
+			return fmt.Errorf("Keccak input has to be felt")
+		}
+		var out [32]byte
+		fp.LittleEndian.PutElement(&out, *v)
+		copy(data[i*25:i*25+25], out[:25]) //25*8 = 200bits
```

Comment:
25 * 8 = 200 bytes ?

Reply:
It's 200 bits. We read 25 bytes from each felt which gives us 200bits per felt. 1600bits in total from the 8 inputs